src/group_theory/quotient_group.lean

```diff
@@ -193,4 +193,44 @@ mul_equiv.of_bijective (ker_lift φ) ⟨ker_lift_injective φ, λ h, begin
   refl
 end⟩
 
+/-- If two normal subgroups `M` and `N` of `G` are the same, their quotient groups are
+isomorphic. -/
+@[to_additive "If two normal subgroups `M` and `N` of `G` are the same, their quotient groups are
+isomorphic."]
+def equiv_quotient_of_eq {M N : subgroup G} [M.normal] [N.normal] (h : M = N) :
+  quotient M ≃* quotient N :=
+{ to_fun := (lift M (mk' N) (λ m hm, quotient_group.eq.mpr (by simpa [← h] using M.inv_mem hm))),
+  inv_fun := (lift N (mk' M) (λ n hn, quotient_group.eq.mpr (by simpa [← h] using N.inv_mem hn))),
+  left_inv := λ x, x.induction_on' $ by { intro, refl },
+  right_inv := λ x, x.induction_on' $ by { intro, refl },
+  map_mul' := λ x y, by rw map_mul }
+
+  section snd_isomorphism_thm
+
+open subgroup
+
+/-- The second isomorphism theorem: given two subgroups `H` and `N` of a group `G`, where `N`
+is normal, defines an isomorphism between `H/(H ∩ N)` and `(HN)/N`. -/
+@[to_additive "The second isomorphism theorem: given two subgroups `H` and `N` of a group `G`,
+where `N` is normal, defines an isomorphism between `H/(H ∩ N)` and `(H + N)/N`"]
+noncomputable def quotient_inf_equiv_prod_normal_quotient (H N : subgroup G) [N.normal] :
+  quotient ((H ⊓ N).comap H.subtype) ≃* quotient (N.comap (H ⊔ N).subtype) :=
+/- φ is the natural homomorphism H →* (HN)/N. -/
+let φ : H →* quotient (N.comap (H ⊔ N).subtype) :=
+  (mk' $ N.comap (H ⊔ N).subtype).comp (inclusion le_sup_left) in
+have φ_surjective : function.surjective φ := λ x, x.induction_on' $
+  begin
+    rintro ⟨y, (hy : y ∈ ↑(H ⊔ N))⟩, rw mul_normal H N at hy,
+    rcases hy with ⟨h, n, hh, hn, rfl⟩,
+    use [h, hh], apply quotient.eq.mpr, change h⁻¹ * (h * n) ∈ N,
+    rwa [←mul_assoc, inv_mul_self, one_mul],
+  end,
+suffices h : (H ⊓ N).comap H.subtype = φ.ker,
+  from (equiv_quotient_of_eq h).trans (quotient_ker_equiv_of_surjective φ φ_surjective),
+ext $ λ x,
+  ⟨λ h, by { simpa [←comap_ker, ker_mk] using h.2 },
+  λ h, by { use x.2, simpa [←comap_ker, ker_mk, mem_comap] using h }⟩
```

Comment:
The goal state halfway through this proof is:
```
begin
  dsimp [φ],
  rw [←comap_ker, ker_mk],
  sorry  -- comap H.subtype (H ⊓ N) = comap (inclusion le_sup_left) (comap (H ⊔ N).subtype N)
end
```
Should there be some `simp` lemmas about `comap (inclusion)` such that this goal can be closed without needing `ext`?

Reply:
Perhaps something like this?
```lean
lemma comap_inclusion_eq_inf {H K L : subgroup G} (hH : H ≤ L) (hK : K ≤ L) :
  comap (inclusion hH) (comap L.subtype K) = comap H.subtype (H ⊓ K) :=
ext $ λ x, by { have hx : ↑x ∈ H := x.prop, simp [hx] }
```